tests/integration/buildcmd/test_build_cmd.py

```diff
@@ -1837,3 +1838,56 @@ def test_nested_build(self, use_container, cached, parallel):
                     ("LocalNestedStack/Function2", {"pi": "3.14"}),
                 ],
             )
+
+
+@skipIf(
+    ((IS_WINDOWS and RUNNING_ON_CI) and not CI_OVERRIDE),
+    "Skip build tests on windows when running in CI unless overridden",
+)
+class TestBuildWithCustomBuildImage(BuildIntegBase):
+    template = "provided_image_function.yaml"
+
+    @parameterized.expand(
+        [
+            ("use_container", None),
+            ("use_container", "amazon/aws-sam-cli-build-image-nodejs10.x"),
+        ]
+    )
+    @pytest.mark.flaky(reruns=3)
+    def test_custom_build_image_succeeds(self, use_container, build_image):
+        if use_container and SKIP_DOCKER_TESTS:
+            self.skipTest(SKIP_DOCKER_MESSAGE)
+
+        cmdlist = self.get_command_list(use_container=use_container, build_image=build_image)
+
+        LOG.info("Running Command: {}".format(cmdlist))
+        process = Popen(cmdlist, cwd=self.working_dir, stdout=PIPE, stderr=PIPE)
+        try:
+            stdout, stderr = process.communicate(timeout=TIMEOUT)
+            LOG.info(f"Stdout: {stdout.decode('utf-8')}")
+            LOG.info(f"Stderr: {stderr.decode('utf-8')}")
+        except TimeoutExpired:
+            LOG.error(f"Command: {command_list}, TIMED OUT")
+            LOG.error(f"Return Code: {process_execute.returncode}")
+            process.kill()
+            raise
+        process_stderr = stderr.strip()
+
+        self._verify_right_image_pulled(build_image, process_stderr)
+        self._verify_build_succeeds(self.default_build_dir)
+
+        self.verify_docker_container_cleanedup("nodejs10.x")
+
+    def _verify_right_image_pulled(self, build_image, process_stderr):
```

Comment:
I thought we have helper functions that do this already?

Reply:
this is checking the name of the image, I don't think we have anything similar